src/services/codefixes/fixReturnTypeInAsyncFunction.ts

```diff
@@ -39,14 +39,20 @@ namespace ts.codefix {
     });
 
     function getInfo(sourceFile: SourceFile, checker: TypeChecker, pos: number): Info | undefined {
-        const returnTypeNode = getReturnTypeNode(sourceFile, pos);
+        if (isInJSFile(sourceFile)) {
+            return undefined;
+        }
+
+        const token = getTokenAtPosition(sourceFile, pos);
+        const func = findAncestor(token, isFunctionLikeDeclaration);
+        const returnTypeNode = func?.type;
         if (!returnTypeNode) {
             return undefined;
         }
 
         const returnType = checker.getTypeFromTypeNode(returnTypeNode);
         const promisedType = checker.getAwaitedType(returnType) || checker.getVoidType();
-        const promisedTypeNode = checker.typeToTypeNode(promisedType);
+        const promisedTypeNode = checker.typeToTypeNode(promisedType, /*enclosingDeclaration*/ func, /*flags*/ undefined);

```

Comment:
It looks like CFA can't tell `func` is not `undefined` here?

Reply:
That's right - we can tell that `returnTypeNode` is defined but we don't tie that info back.

Reply:
Technically if we removed the check though, we'd be in a bad situation.

Reply:
I was just observing that there seemed to be a potential improvement that could be made to CFA, not suggesting any changes to this code.